### PR TITLE
Changed createCommunity from a page to a dialog box

### PR DESCRIPTION
**Pull request by ZachBourque**
# Description

This PR includes the following proposed change(s):

- Created addCommunityModal.js
- Changed links to /createCommunity to open the modal
- Removed the /createCommunity page

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [x] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/client/src/redux/ducks/communityDuck.js [hunk 1]
```diff
@@ -75,7 +77,7 @@ export const createCommunity =
         body: JSON.stringify({
           title: communityData.title,
           description: communityData.description,
-          creator: communityData.creator,
+          creator: name,
```
Comment (BradyMitch) [inline app/client/src/redux/ducks/communityDuck.js:80]: When you create a community or a post you don't specify the creator in the request body. It grabs that from the authentication

**Comment by BradyMitch** [comment]
I get a communityDuck 400 Bad Request when I try to create a community, but modal is great.

**Comment by BradyMitch** [comment]
> I get a communityDuck 400 Bad Request when I try to create a community, but modal is great.

I think its because rules needs to be specified. For now you can just set it to an empty string.

**Comment by BradyMitch** [comment]
It wont let me pull the changes to the branch. You tested creating a community?

**Comment by ZachBourque** [comment]
> It wont let me pull the changes to the branch. You tested creating a community?

Weird.

I tested, works as expected

**Review by BradyMitch — APPROVED** [review]

**Comment by BradyMitch** [comment]
I fixed the error I was having with not being able to pull from the remote branch by 'git fetch --prune' and then 'git branch -d <branch-name>'. Then I checked out the branch again and your update is working for me.

**Pull request merged**